presto-spi/src/main/java/io/prestosql/spi/type/LongTimestamp.java

```diff
@@ -63,4 +64,14 @@ public int hashCode()
     {
         return Objects.hash(epochMicros, picosOfMicro);
     }
+
+    @Override
+    public int compareTo(LongTimestamp other)
+    {
+        int value = Long.compare(epochMicros, other.epochMicros);
+        if (value != 0) {
+            return value;
+        }
+        return Integer.compareUnsigned(picosOfMicro, other.picosOfMicro);
```

Comment:
no need for unsignedm sine they are in "signed" range

Reply:
This matches the compare implementation in the Type class.